packages/bitcore-client/bin/wallet-sign

```
@@ -1,30 +1,64 @@
+  .option('--txfile [txfile]', 'path to a jsonl file with rawTransaction and utxos')
+  .option('--utxos', 'Pass in utxos to use to sign transaction')
+  .option('--keys [keys]', 'jsonl file of {privKey, pubKey, address} pairs')
   .parse(process.argv);
 
 const main = async () => {
-  const { name, path, tx } = program;
+  const { name, path, tx, txfile, keys, utxos } = program;
   let wallet;
+  let decryptedKeys;
   try {
     const password = await promptly.password('Wallet Password:');
     wallet = await Wallet.loadWallet({ name, path });
     wallet = await wallet.unlock(password);
-    let signedTx = await wallet.signTx({ tx });
-
-    console.log(
-      JSON.stringify({
-        signedTx: signedTx.uncheckedSerialize()
-      })
-    );
+    if (keys) {
+      let contents = fs.readFileSync(keys).toString();
+      decryptedKeys = contents
+        .split('\n')
+        .filter(line => line !== '')
+        .map(line => JSON.parse(line));
+      try {
+        const bitcoinCorePassword = await promptly.password('Enter Bitcoin Core Password:');
+        let decryptedParams = Encryption.bitcoinCoreDecrypt(decryptedKeys, bitcoinCorePassword);
+        decryptedKeys = decryptedParams.jsonlDecrypted;
+      } catch (e) {
+        console.error(e);
+      }
+    }
+    if (tx) {
+      let signedTx = await wallet.signTx({ tx, keys: decryptedKeys, utxos });
+      console.log(JSON.stringify(signedTx));
+    }
+    if (txfile) {
+      const readStream = fs.createReadStream(txfile, {flags: 'r', encoding: 'utf8'});
+      const writeStream = fs.createWriteStream('signedTx.jsonl');
+      readStream
+        .pipe(StreamUtil.jsonlBufferToObjectMode()).on('error', (err) => console.error(err))
+        .pipe(StreamUtil.signTxStream(wallet, decryptedKeys, utxos)).on('error', (err) => console.error(err))
+        .pipe(StreamUtil.objectModeToJsonlBuffer()).on('error', (err) => console.error(err))
+        .pipe(writeStream).on('error', (err) => console.error(err));
```

Is it possible to just do one `on('error', console.error)` at the end of these pipes?

---

I believe errors are not propagated through streams. You would need to add an error listener to every stream